From 1b2eccc8a3c8c70855973d9dcff46bb0c4f79345 Mon Sep 17 00:00:00 2001
From: Michael Marineau <mike@marineau.org>
Date: Mon, 12 Aug 2013 17:42:39 -0400
Subject: [PATCH] Fix test using ? glob with ObjectToURI

Passing a ? wildcard through ObjectToURI does not work because passing
the constructed URI through urlparse.urlparse() interprets ? as a query
and strips it off since there is no query string following it. The *
glob which is actually used in the very next call doesn't have that
problem so just use that instead.
---
 gslib/tests/test_naming.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gslib/tests/test_naming.py b/gslib/tests/test_naming.py
index b4d0ccd..8208238 100755
--- a/gslib/tests/test_naming.py
+++ b/gslib/tests/test_naming.py
@@ -840,7 +840,7 @@ class GsutilNamingTests(testcase.GsUtilUnitTestCase):
     for i, final_dst_char in enumerate(('', '/')):
       # Copy some files into place in dst bucket.
       self.RunCommand(
-          'cp', [suri(src_bucket_uri, 'f%df?' % i),
+          'cp', [suri(src_bucket_uri, 'f%df*' % i),
                  suri(dst_bucket_uri, 'dst_subdir%d' % i) + final_dst_char])
       # Now do the move test.
       self.RunCommand(
-- 
1.8.1.5

